project1-sps/script.js

```diff
@@ -0,0 +1,155 @@
+var counterForComputerWin = 0;
+
+// main function to compare computer's decision v.s. human input
+var main = function (input) {
+  var computerDecision = convertRandomNumberToSpsState();
+  // var computerMode= mainMode(input1);
+
+  console.log("computer decides to show " + computerDecision);
+  console.log("You showed " + input);
+
+  // conditional check for draw
+  if (input == "scissors" && computerDecision == "scissors") {
+    var result = `It's a draw. You showed ${input}, and computer also showed ${computerDecision}`;
+    return result;
+  }
+  if (input == "paper" && computerDecision == "paper") {
+    var result = `It's a draw. You showed ${input}, and computer also showed ${computerDecision}`;
+    return result;
+  }
+  if (input == "stone" && computerDecision == "stone") {
+    var result = `It's a draw. You showed ${input}, and computer also showed ${computerDecision}`;
+    return result;
+  }
+
+  // conditional check if you win
+  if (input == "paper" && computerDecision == "stone") {
+    counterForYouWin = counterForYouWin + 1;
+    console.log(
+      `you wins ${counterForYouWin}x, computer won ${counterForComputerWin}x`
+    );
+
+    var result = `You win! Your ${input} beats computer's ${computerDecision}`;
+    return result;
+  }
+
+  if (input == "stone" && computerDecision == "scissors") {
+    counterForYouWin = counterForYouWin + 1;
+    console.log(
+      `you wins ${counterForYouWin}x, computer won ${counterForComputerWin}x`
+    );
+
+    var result = `You win! Your ${input} beats computer's ${computerDecision}`;
+    return result;
+  }
+
+  if (input == "scissors" && computerDecision == "paper") {
+    counterForYouWin = counterForYouWin + 1;
+    console.log(
+      `you wins ${counterForYouWin}x, computer won ${counterForComputerWin}x`
+    );
+
+    var result = `You win! Your ${input} beats computer's ${computerDecision}`;
+    return result;
+  }
+
+  // conditional check if you lose
+  if (input == "stone" && computerDecision == "paper") {
+    counterForComputerWin = counterForComputerWin + 1;
+    console.log(
+      `computer wins ${counterForComputerWin}x, you won ${counterForYouWin}x`
+    );
+
+    var result = `Oops, you lose! Your ${input} is beaten by computer's ${computerDecision}`;
+    return result;
+  }
+  if (input == "scissors" && computerDecision == "stone") {
+    counterForComputerWin = counterForComputerWin + 1;
+    console.log(
+      `computer wins ${counterForComputerWin}x, you won ${counterForYouWin}x`
+    );
+
+    var result = `Oops, you lose! Your ${input} is beaten by computer's ${computerDecision}`;
+    return result;
+  }
+
+  if (input == "paper" && computerDecision == "scissors") {
+    counterForComputerWin = counterForComputerWin + 1;
+    console.log(
+      `computer wins ${counterForComputerWin}x, you won ${counterForYouWin}x`
+    );
+    var result = `Oops, you lose! Your ${input} is beaten by computer's ${computerDecision}`;
+    return result;
+  }
+};
```

Comment:
these can be combined tgt too, but recall my earlier comment about not having to check for the last scenario? it applies here as well! in SPS theres only 3 outcomes - tie, player wins, or player loses. if our code reaches here after we've checked for if its a tie or if player won, then we know that player lost.

so you can just increment computer win count and return your string here without any `if` conditionals